### PR TITLE
Add prettier

### LINKED ISSUE #149: Add Prettier formatter to linter once they add a no-semicolon option
https://github.com/prettier/prettier
https://github.com/not-an-aardvark/eslint-plugin-prettier

### PR DESCRIPTION
**Pull request by trevorgerhardt**
I meant to separate out the formatting from the feature but it's all in that first commit 😬 

Closes #149

### CODE REVIEW

File: lib/lint-messages.js [hunk 1]
```diff
@@ -57,11 +57,21 @@ function lintFileContents (messages, js) {
     lineNumber++
     const parsedImport = parseImportLine(line)
     if (parsedImport) {
-      [importedAtRootAs, importedMembersAs] = parsedImport
+      ;[importedAtRootAs, importedMembersAs] = parsedImport
```
Comment (evansiroky) [inline lib/lint-messages.js]: Interesting addition of a semicolon here.

**Comment by evansiroky** [comment]
What has prompted us to switch to not having a space after the function keyword?

File: lib/lint-messages.js (hunk 1 above)
Comment (trevorgerhardt) [inline lib/lint-messages.js]: That's the standard use case for advocating pro-semicolons. For example:

```js
const a = b
[0].map(doSomething)
```
Would be interpreted as
```js
const a = b[0].map(doSomething)
```

Comment (evansiroky) [inline lib/lint-messages.js]: Ah, I see

**Comment by trevorgerhardt** [comment]
@evansiroky Prettier formats it that way. It's the only difference between the standard linter and the way prettier formats now and I don't lean hard either way on this rule so I just relaxed it in our eslint config. See here: https://github.com/prettier/prettier/issues/1139

**Comment by evansiroky** [comment]
I also don't lean hard either way, but it did cause lots of changes in the code style.

**Comment by mattwigway** [comment]
Does that only affect anonymous functions?

File: lib/browserify.js [hunk 2]
```diff
@@ -6,12 +6,7 @@ const transform = require('./js-transform')
 
 module.exports = browserifyIt
 
-function browserifyIt ({
-  config,
-  entry,
-  env,
-  minify
-}) {
+function browserifyIt({config, entry, env, minify}) {
```
Comment (evansiroky) [inline lib/browserify.js]: @mattwigway no

Comment (mattwigway) [inline lib/browserify.js]: Oh, I feel kinda strongly about that, it makes it look really cramped.

Comment (trevorgerhardt) [inline lib/browserify.js]: *compact 😉

Comment (mattwigway) [inline lib/browserify.js]: The seats on United are so compact that sometimes they have to re-accomodate passengers.

Comment (trevorgerhardt) [inline lib/browserify.js]: Luckily our code isn't in a rush to get anywhere

Comment (trevorgerhardt) [inline lib/browserify.js]: Btw, I agree in that I'd rather not change this particular rule across our code base. But I also would rather use a standard formatter much more than have spaces before function names.

Comment (mattwigway) [inline lib/browserify.js]: I'm not going to block it over this, yeah. Although I do question whether we want something that completely rewrites all our code.

Comment (trevorgerhardt) [inline lib/browserify.js]: Why is that?

**Comment by trevorgerhardt** [comment]
Closing in favor of #183

**Comment by trevorgerhardt** [comment]
@evansiroky anything else we should add to this?

**Comment by evansiroky** [comment]
I think it's good to go.

**Pull request merged**